api/golang/engine/lib/kurtosis_context/portal_daemon_client.go

```diff
@@ -0,0 +1,66 @@
+package kurtosis_context
+
+import (
+	"context"
+	"fmt"
+	portal_constructors "github.com/kurtosis-tech/kurtosis-portal/api/golang/constructors"
+	portal_api "github.com/kurtosis-tech/kurtosis-portal/api/golang/generated"
+	contexts_store_api "github.com/kurtosis-tech/kurtosis/contexts-config-store/api/golang"
+	contexts_store_generated_api "github.com/kurtosis-tech/kurtosis/contexts-config-store/api/golang/generated"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+const (
+	DefaultGrpcPortalClientPortNum = uint16(9731)
+)
+
+// BuildPortalDaemonClient builds a portal daemon GRPC client based on the current context and a
+// forLocalContextReturnNilIfUnreachable flag
+// If the flag is set to true, it returns a nil client for local contexts when the portal is not reachable.
+// Local context are working seamlessly without the portal (unless the user wants to map port). Therefore we're not
+// requiring it for most workflows
+// TODO: when it's assumed that Kurtosis Portal should be running everywhere, we can remove this logic and just build a
+// regular client
+func BuildPortalDaemonClient(currentContext *contexts_store_generated_api.KurtosisContext, forLocalContextReturnNilIfUnreachable bool) (portal_api.KurtosisPortalClientClient, error) {
```

Comment:
just for curiosity and learning purposes, Why you decided to call it `Build` and not `New` or `Create`?

Reply:
Good call here, I guess I didn't think too much about the name of this function. Based on what we already discussed in previous reviews, I'll rename it `Create*`

Reply:
(resolving to unblock merging but happy to discuss more it you think it's worth it)